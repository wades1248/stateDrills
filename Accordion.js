import React from 'react';

class Accordion extends React.Component {
    static defaultProps = {
        sections: []
    }
    state ={
        currentIndex: null
    }
    handleClick(index){
        this.setState({currentIndex: index})
    }   
    renderSections() {
        return(
            this.props.sections.map((section, index)=> {                    return(
                    <li key={index}>
                        <button onClick={() =>this.handleClick(index)}>
                            {section.title}
                        </button>
                        {/*this.state.currentIndex===index && <p>{section.content}</p>*/}
                        {this.renderSelected(section,index)}
                    </li>
                )
            })
        )
        
        
    }
    renderSelected(section, index){
        if(this.state.currentIndex===index){
            return(
                <p>{section.content}/</p>

            )
        }
    }
    render(){
        return (
        <ul>
            {this.renderSections()}
        </ul>
    )}
}

export default Accordion;