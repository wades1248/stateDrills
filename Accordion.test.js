import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from'enzyme-to-json';
import Accordion from './Accordion';
import { exportAllDeclaration } from '@babel/types';

describe('Accordion Component', () => {
    const sectionsProp = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]

      it('renders without errors', () =>{
          const div= document.createElement('div');
          ReactDOM.render(<Accordion />, div);
          ReactDOM.unmountComponentAtNode(div);
      })
      it('renders empty given no sections prop', ()=>{
          const wrapper = shallow(<Accordion />)
          expect(toJson(wrapper)).toMatchSnapshot()
      })
      it('renders no sections as active by default', ()=>{
          const wrapper = shallow(<Accordion sections={sectionsProp} />)
          expect(toJson(wrapper)).toMatchSnapshot()
      })
      it('opens the clicked section', () => {
          const wrapper = shallow(<Accordion sections={sectionsProp} />)
          wrapper.find('button').at(1).simulate('click')
          expect(toJson(wrapper)).toMatchSnapshot()
      } )
      it('closes previously selected sections whena new section is selected', () =>{
        const wrapper = shallow(<Accordion sections={sectionsProp} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
      })

})