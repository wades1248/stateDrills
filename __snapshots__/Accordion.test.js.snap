// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes previously selected sections whena new section is selected 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      /
    </p>
  </li>
</ul>
`;

exports[`Accordion Component opens the clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      /
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no sections prop 1`] = `<ul />`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
